component/conprof/http/api.go

```diff
@@ -301,17 +301,24 @@ func queryAndDownload(c *gin.Context) error {
 	fn := func(pt meta.ProfileTarget, ts int64, data []byte) error {
 		fileName := fmt.Sprintf("%v_%v_%v_%v", pt.Kind, pt.Component, pt.Address, ts)
 		fileName = strings.ReplaceAll(fileName, ":", "_")
-		if param.DataFormat == meta.ProfileDataFormatSVG {
-			svg, err := ConvertToSVG(data)
-			if err == nil {
-				data = svg
-				fileName += ".svg"
-			}
-		}
 		if pt.Kind == meta.ProfileKindGoroutine {
 			fileName += ".txt"
+		} else {
+			if param.DataFormat == meta.ProfileDataFormatSVG {
+				svg, err := ConvertToSVG(data)
+				if err == nil {
+					data = svg
+					fileName += ".svg"
+				}
```

Comment:
When will it happen?

Reply:
What do you mean? Normally the `err` here should always be nil. If the `data` was damaged by unknow reason, the err will happen.

Reply:
I mean this `param.DataFormat == meta.ProfileDataFormatSVG` might not be needed any more, since TiDB Dashboard will not use in this way I guess? @baurine

Reply:
> I mean this `param.DataFormat == meta.ProfileDataFormatSVG` might not be needed any more, since TiDB Dashboard will not use in this way I guess? @baurine

Currently, we get the graph(not flamegraph) from ngm directly, because the dashboard  is just a simple proxy, we don't get the protobuf and convert it to SVG in dashboard side.

Reply:
Got it.

Reply:
Oh, you're right, for download, we won't request SVG, whatever single or bundle. @breeswish , we only request SVG when viewing the graph.

Reply:
So looks like you should remove the param.DataFormat I guess @crazycs520